assignment4.Rmd

```diff
@@ -105,14 +132,45 @@ sweep(loadings, 2, colSums(loadings), "/") #sweep() computes each row as a propo
 
 #Now examine your components and try to come up with substantive descriptions of what some might represent?
 
+# PC1 has higher influence score in mean_hint, problems_attempted and mean_attempt, which might represent the status of student in current session.
+# PC2 has higher influence score in prior_percent_correct and prior_prob_count, which might represent the status of student in prior session.
+# PC3 has higher influence score in mean_confidence, mean_attempt and prior_prob_count, which might represent the confidence level of student after finishing prior questions.
+# PC4 has higher influence score in prior_prob_count and mean_confidence, which might represent the completion level of problems during the whole session.
+# PC5 has higher influence score in mean_attempt and problems_attempted, which might represent the effort of student.
+# PC6 has the the highest influence score mean_hint, which might represent the mastery of knowledge of student in current session. 
+
+prop.table(x = loadings, margin = 2)
+  
 #You can generate a biplot to help you, though these can be a bit confusing. They plot the transformed data by the first two components. Therefore, the axes represent the direction of maximum variance. Then mapped onto this point cloud are the original directions of the variables, depicted as red arrows. It is supposed to provide a visualization of which variables "go together". Variables that possibly represent the same underlying construct point in the same direction.  
 
 biplot(pca)
 
 #Calculate values for each student that represent these your composite variables and then create a new correlogram showing their relationship to mean_correct.
 
+# multiply pca$x by transpose of pca$rotation to generate composite variables
+C1 <- pca$x %*% t(pca$rotation)
+C2 <- cbind(C1,as.data.frame(D1$mean_correct))
+corrgram(C2, order=TRUE, lower.panel=panel.shade,
+  upper.panel=panel.pie, text.panel=panel.txt)
 ```
 # Part III  
 
 ## Also in this repository is a data set collected from class that shows how many students thought that a TC program was related to andother TC program. Use PCA to look for components of related programs. Explain why you think there are relationships between these programs.
 
+```{r}
+T1 <- read.csv("TCprograms.csv", sep = " ")
+rownames(T1) <- T1$program
+T1 <- T1[,-1]
+T1 <- T1[-69,] # delete blank row
+T2 <- scale(T1, center = TRUE)
+pca_tc <- prcomp(T2, scale = TRUE)
+summary(pca_tc)
+biplot(pca_tc,cex = 0.5)
+
+# The graph in the top right has word "psychology" most frequently, and bottom left part has word "education" most frequently. So I expect program has word "psychology" has relationship with each other, and program has word "education" has relationship with each other.
+
+
+# loadings2 <- abs(pca_tc$rotation)
+# loadings3 <- data.frame(prop.table(x = loadings2, margin = 2))[,1:4]
+```
```

Comment:
Overall, excellent work but you might want to be more specific in the Part III analysis write-up.